### PR TITLE
GEODE-9327: Remove ACE networking code references

### PR DESCRIPTION
**Pull request by gaussianrecurrence**
- Replaced method implementation to get a random port in old ITs framework.
 - Replaced convertHostToCanonicalForm implementation.
 - Changed ClientProxyMembershipID address serialization.
 - Removed unused PoolFactory::addCheck method.
 - Modified QueueConnectionRequestTest test to pass after the changes.

### CODE REVIEW

**Comment by pdxcodemonkey** [comment]
@gaussianrecurrence Greetings!  Are you still working on any of these PRs?  If not, may we copy your WIP on the ACE PRs to another fork and continue the work?

**Comment by gaussianrecurrence** [comment]
> @gaussianrecurrence Greetings! Are you still working on any of these PRs? If not, may we copy your WIP on the ACE PRs to another fork and continue the work?

Sadly @pdxcodemonkey, I am currently working on urgent issues right now, so won't be able to work on this on the short-term. So feel free to move to it to another fork to continue the work. As soon as I find the time, I'll be more than happy to help with this work :)

File: cppcache/src/ClientProxyMembershipID.cpp [hunk 1]
```diff
@@ -77,21 +77,13 @@ ClientProxyMembershipID::ClientProxyMembershipID(
...
-    const auto socketAddress6 =
-        static_cast<const struct sockaddr_in6*>(address.get_addr());
-    auto socketAddress =
-        reinterpret_cast<const uint8_t*>(&socketAddress6->sin6_addr);
-    auto length = sizeof(socketAddress6->sin6_addr);
-    m_hostAddr.assign(socketAddress, socketAddress + length);
+    const boost::asio::ip::address& address) {
+  if (address.is_v6()) {
+    auto bytes = address.to_v6().to_bytes();
+    m_hostAddr.assign(bytes.begin(), bytes.end());
```
Comment (pdxcodemonkey) [inline cppcache/src/ClientProxyMembershipID.cpp:83]: This is probably as good a time as any to rename member variables...

File: cppcache/src/Utils.cpp [hunk 2]
```diff
@@ -81,43 +83,36 @@ void Utils::parseEndpointString(const char* endpoints, std::string& host,
...
+  io_service svc;
+  boost::system::error_code ec;
+  tcp::resolver resolver{svc};
+  auto results = resolver.resolve(hostname, ec);
+
+  if (!ec) {
+    hostname = results->host_name();
+  }
+
+  return hostname + port;
```
Comment (pdxcodemonkey) [inline cppcache/src/Utils.cpp:115]: Shouldn't this be `hostname + ":" + port`?

**Review by pdxcodemonkey — CHANGES_REQUESTED** [review]

File: cppcache/src/ClientProxyMembershipID.cpp (hunk 1 above)
Comment (gaussianrecurrence) [reply cppcache/src/ClientProxyMembershipID.cpp:83]: Good point!

File: cppcache/src/Utils.cpp (hunk 2 above)
Comment (gaussianrecurrence) [reply cppcache/src/Utils.cpp:115]: When re-checking this code I made myself this question. Thing is port is already containing the semicolon as of the way the endpoint is split. However, I'll change it so port does not contain the semicolon in order to make the code more readable.

**Review by pdxcodemonkey — APPROVED** [review]

**Pull request merged**